app/models/event.rb

```diff
@@ -660,7 +660,7 @@ def change_status_and_notify
         end
 
         # Check if the user is eligible for role change
-        if self.user.approved_events_count >= User::EVENT_APPROVAL_THRESHOLD &&
+        if self.user.approved_events_count > User::EVENT_APPROVAL_THRESHOLD &&
```

Comment:
in this workflow the event approval count is updated before the role upgrade happens

the logic
- so if the user had 2 event approved
- after running this line self.user.update!(approved_events_count: self.user.approved_events_count + 1)
- the event approval would be 3 now, but as per documentation 
- `We need to manually handle the change of roles from default or basic user to trusted user through admin console if we see the number of successfully published event > 3`
- though it is written in documentation that `We need to manually handle the change of roles from default or basic user to trusted user` but i have discussion with @harshita3199  and after discussion we agree to automatic update

Reply:
ok, then the requirements are unclear to me, and this should be clarified with @kristen-schroeder as the product owner. 
@kristen-schroeder do we want to have the manual change to trusted user or an automated one? and does the user need a minimum of 3 or a minimum of 4 approved events to become a trusted one?

Reply:
> ok, then the requirements are unclear to me, and this should be clarified with @kristen-schroeder as the product owner. @kristen-schroeder do we want to have the manual change to trusted user or an automated one? and does the user need a minimum of 3 or a minimum of 4 approved events to become a trusted one?

after 3 successfully approved events, a user shall be upgraded to a trusted user

i have less clarity on the discussion of automatic vs. manual upgrading of a user, because the best system depends on our users behaviour. there is a fear of a potentially misbehaving trusted user, and we have no data yet to see if this is founded or not. so either we structure it as:

- users are automatically upgraded to trusted user after 3 successful events AND trusted users can be demoted to user by admin

or

- manual upgrading of everyone to trusted user

i favour the first one.

Reply:
As per our discussion earlier, I also favour the first one hence decided to go with the automated upgrade as during my discussion with @adeel-ahmad-awan we discussed how complex it would be to achieve and it was achievable so implemented automated one. I will also update the workflow with the changes.

Reply:
at @kristen-schroeder  just to be sure, when you say 

> after 3 successfully approved events, a user shall be upgraded to a trusted user

in the code implementation when the forth event is approved the user role is changed. As per the documentation we have successfully published event > 3 (successfully published event grater then 3)

Reply:
> at @kristen-schroeder just to be sure, when you say
> 
> > after 3 successfully approved events, a user shall be upgraded to a trusted user
> 
> in the code implementation when the forth event is approved the user role is changed. As per the documentation we have successfully published event > 3 (successfully published event grater then 3)

that works! :D

Reply:
thanks everyone. sounds like `>= 3` then after all.

Reply:
@umunixie, I believe there may have been a slight misunderstanding. 

Kristen’s response, “that works!”, was in reply to the suggestion of upgrading the user when the count is > 3 (on the 4th approved event). It seems there might have been an interpretation that the decision was to use >= 3 instead. Since I already have a meeting scheduled with @harshita3199, perhaps we could briefly discuss this together to ensure we’re all on the same page?

Reply:
@adeel-ahmad-awan i think i replied before github showed me the newest comments submitted before mine. anyway, my thinking goes like this: when i see "threshold for becoming a trusted user is 3 approved events", i interpret it as "for the 4th event approval is not needed anymore". maybe it's just how i read things, but it also feels more intuitive for me to have a `>= 3` form in the code because ` > 3` forces me to _think_ (what's greater than 3? ah right, 4) while `>= 3` doesn't force me to think, it allows me to just _see_.

but this is not a blocker for merging as long as it works for @kristen-schroeder with 4 approved events.

Reply:
@umunixie since this point was discussed in the meeting, can we close this conversation

Reply:
@adeel-ahmad-awan so you don't agree with the code readability aspect?

Reply:
I think this comes down to personal preference and how we logically interpret conditions.

For me, > 3 aligns with the idea that the threshold is crossed only after exceeding three. It clearly indicates that the action will trigger when the count goes beyond three, which reflects our requirements.

Additionally, since the value 3 is stored in a constant `EVENT_APPROVAL_THRESHOLD`, using `> EVENT_APPROVAL_THRESHOLD` keeps the logic consistent. If we were to set the constant to 4 and change the condition to `>= 4,` it might cause unnecessary confusion.

Reply:
agree to disagree then :)